esmvalcore/preprocessor/_io.py

```diff
@@ -179,6 +179,41 @@ def concatenate(cubes):
     raise ValueError(f'Can not concatenate cubes: {msg}')
 
 
+def concatenate(cubes):
+    """Concatenate all cubes after fixing metadata."""
+    _fix_cube_attributes(cubes)
+
+    if len(cubes) > 2:
+        cubes_dict = {cube: cube.coord("time").points[0] for cube in cubes}
+        cubes_dict = OrderedDict(
+            {k: v for k, v in sorted(cubes_dict.items(),
+                                     key=lambda item: item[1])}
+        )
+        cubes = [item[0] for item in cubes_dict.items()]
+        concatenated = iris.cube.CubeList([cubes[0], cubes[1]]).concatenate()
+        if len(concatenated) == 2:
+            concatenated = _by_two_concatenation(concatenated)
+        else:
+            concatenated = concatenated[0]
+        for i in range(2, len(cubes)):
+            concatenated = [concatenated, cubes[i]]
+            concatenated = iris.cube.CubeList(concatenated).concatenate()
+            if len(concatenated) == 2:
+                concatenated = _by_two_concatenation(concatenated)
+            else:
+                concatenated = concatenated[0]
+    else:
+        concatenated = iris.cube.CubeList(cubes).concatenate()
+        if len(concatenated) == 2:
+            concatenated = _by_two_concatenation(cubes)
+
+    if isinstance(concatenated, iris.cube.CubeList):
+        concatenated = concatenated[0]
```

Comment:
Can you try to simplify this logic into a single small loop? I think that all you need is something like
```python
result = cubes[0]
for cube in cubes[1:]:
    result = _by_two_concatenation(result, cube)
```

Reply:
yip, just done it, have a look